### PR TITLE
Add Java option to disable capturing command line

### PR DESCRIPTION
**Pull request by mikezhang1234567890**
Signed-off-by: Mike Zhang <mike.h.zhang@ibm.com>

fyi @pshipton 

Simple test file `PrintEnv.java`: 
```java
import java.util.Map;

public class PrintEnv {
    public static void main(String[] a) {
        Map<String, String> env = System.getenv();
        for (Map.Entry<String, String> entry : env.entrySet()) {
            String line = (entry.getKey() + "=" + entry.getValue()).replace("\r\n", "").replace("\n", "");
            System.out.println(line);
        }
    }
}
```

Output:
```
$ $TEST_JDK_BIN/java PrintEnv.java | grep OPENJ9
OPENJ9_JAVA_COMMAND_LINE=/Users/mike-z/scratch/ext-jdk18/build/macosx-x86_64-server-release/images/jdk/bin//java

$ $TEST_JDK_BIN/java -XX:-OpenJ9CommandLineEnv PrintEnv.java | grep OPENJ9

$ $TEST_JDK_BIN/java -XX:+OpenJ9CommandLineEnv PrintEnv.java | grep OPENJ9
OPENJ9_JAVA_COMMAND_LINE=/Users/mike-z/scratch/ext-jdk18/build/macosx-x86_64-server-release/images/jdk/bin//java
```

### CODE REVIEW

**Comment by pshipton** [comment]
Initial review indicates this solution doesn't resolve the problem we were trying to fix. I'll set this as draft until we can update it.

File: runtime/j9vm/jvm.c [hunk 1]
```diff
@@ -1540,6 +1540,7 @@ typedef struct J9SpecialArguments {
 	UDATA *argEncoding;
 	IDATA *ibmMallocTraceSet;
 	const char *executableJarPath;
+	BOOLEAN noCaptureCommandline;
```
Comment (keithc-ca) [inline runtime/j9vm/jvm.c:1543]: I think the code would read better if this were not negated:
```
    BOOLEAN captureCommandline;
```

File: runtime/oti/jvminit.h [hunk 2]
```diff
@@ -391,6 +391,7 @@ enum INIT_STAGE {
 #define VMOPT_XXDISABLEUTFCACHE "-XX:-UTFCache"
 #define VMOPT_XXENABLEENSUREHASHED "-XX:+EnsureHashed:"
 #define VMOPT_XXDISABLEENSUREHASHED "-XX:-EnsureHashed:"
+#define VMPOT_XXNOOPENJ9COMMANDLINEENV "-XX:NoOpenJ9CommandLineEnv"
```
Comment (keithc-ca) [inline runtime/oti/jvminit.h:394]: Perhaps this should be `-XX:-OpenJ9CommandLineEnv` for consistency?

**Review by keithc-ca — CHANGES_REQUESTED** [review]
The title does not match the intent of this.

File: runtime/oti/jvminit.h (hunk 2 above)
Comment (pshipton) [reply runtime/oti/jvminit.h:394]: Agree. And in that case we should support -XX:+OpenJ9CommandLineEnv as well.

**Review by pshipton — APPROVED** [review]

**Comment by mikezhang1234567890** [comment]
Fixed some spacing in https://github.com/eclipse-openj9/openj9/commit/a9507abddf72b497736be6d7137f2a888d9c0658

**Review by keithc-ca — APPROVED** [review]

**Comment by keithc-ca** [comment]
jenkins test sanity alinux64 jdk17

**Pull request merged**